src/data_info/services/stat.clj

```diff
@@ -46,76 +62,107 @@
 
 
 (defn- merge-counts
-  [stat-map cm user path]
-  (if (is-dir? stat-map)
+  [stat-map cm user path included-keys]
+  (if (and (needs-any-key? included-keys :file-count :dir-count) (is-dir? stat-map))
     (assoc stat-map
-      :file-count (icat/number-of-files-in-folder user (cfg/irods-zone) path)
-      :dir-count  (icat/number-of-folders-in-folder user (cfg/irods-zone) path))
+      :file-count (if (needs-key? included-keys :file-count) (icat/number-of-files-in-folder user (cfg/irods-zone) path))
+      :dir-count  (if (needs-key? included-keys :dir-count)  (icat/number-of-folders-in-folder user (cfg/irods-zone) path)))
     stat-map))
 
 
 (defn- merge-shares
-  [stat-map cm user path]
-  (if (owns? stat-map)
+  [stat-map cm user path included-keys]
+  (if (and (needs-key? included-keys :share-count) (owns? stat-map))
     (assoc stat-map :share-count (count-shares cm user path))
     stat-map))
 
 
 (defn- merge-label
-  [stat-map user path]
-  (assoc stat-map
-         :label (paths/path->label user path)))
+  [stat-map user path included-keys]
+  (if (needs-key? included-keys :label)
+    (assoc stat-map
+           :label (paths/path->label user path))
+    stat-map))
 
 
 (defn- merge-type-info
-  [stat-map cm user path]
-  (if-not (is-dir? stat-map)
+  [stat-map cm user path included-keys]
+  (if (and (needs-any-key? included-keys :infoType :content-type) (not (is-dir? stat-map)))
     (assoc stat-map
-      :infoType     (get-types cm user path)
-      :content-type (irods/detect-media-type cm path))
+      :infoType     (if (needs-key? included-keys :infoType) (get-types cm user path))
+      :content-type (if (needs-key? included-keys :content-type) (irods/detect-media-type cm path)))
     stat-map))
 
-
 (defn ^IPersistentMap decorate-stat
-  [^IPersistentMap cm ^String user ^IPersistentMap stat]
+  [^IPersistentMap cm ^String user ^IPersistentMap stat included-keys]
```

Comment:
IDE is telling me that this change will break `data-info.services.path-lists/create-path-list`.
It looks like `data-info.services.uuids/paths-for-uuids` will break as well (although that's not currently in use anywhere).